assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js

```diff
@@ -66,36 +66,69 @@ export default function Idea( props ) {
 		dismissIdea,
 		setActivity,
 		removeActivity,
+		removeIdeaFromNewIdeas,
 		removeIdeaFromNewAndSavedIdeas,
+		moveIdeaFromSavedIdeasToNewIdeas,
+		moveIdeaFromNewIdeasToSavedIdeas,
 	} = useDispatch( MODULES_IDEA_HUB );
 
 	const activity = useSelect( ( select ) =>
 		select( MODULES_IDEA_HUB ).getActivity( name )
 	);
 
 	const handleDelete = useCallback( async () => {
-		setActivity( name, IDEA_HUB_ACTIVITY_IS_PROCESSING );
+		setActivity( name, IDEA_HUB_ACTIVITY_IS_DELETING );
 		await dismissIdea( name );
-		removeActivity( name );
+		setActivity( name, IDEA_HUB_ACTIVITY_DELETED );
```

Comment:
Now we have separate activities per action and can move `setActivity` calls to the corresponding actions:

```diff
 	*dismissIdea( ideaName ) {
 		invariant( typeof ideaName === 'string', 'ideaName must be a string.' );

 		yield clearError( 'dismissIdea', [ ideaName ] );
+		yield baseActions.setActivity( ideaName, IDEA_HUB_ACTIVITY_IS_DELETING );

 		const { response, error } = yield baseActions.updateIdeaState( {
 			name: ideaName,
 			dismissed: true,
 		} );

 		if ( error ) {
 			yield receiveError( error, 'dismissIdea', [ ideaName ] );
 		}

+		yield baseActions.setActivity( ideaName, IDEA_HUB_ACTIVITY_DELETED );

 		return { response, error };
 	},
```

Same for other activities.

Reply:
@eugene-manuilov this looks like a great enhancement 👍  Would we also need to set/restore the activity if there was an error updating the idea state?

Reply:
@aaemnnosttv @eugene-manuilov I think it'll be good if we reset the activity in case there's an error. I've updated the PR to cater for that.